src/core/Form/FilterInput/FilterInput.tsx

```diff
@@ -0,0 +1,186 @@
+   * @default default
+   */
+  status?: FilterInputStatus;
+  /** Status text to be shown below the component and hint text. Use e.g. for validation error */
+  statusText?: string;
+  /** FilterInput name */
+  name?: string;
+  /** Align label on top or on the left side of the input field
+   * @default 'top'
+   */
+  labelAlign?: 'top' | 'left';
+  /** Items to be filtered */
+  items: Array<T>;
+  /** Returns the filtered items */
+  onFiltering: (filteredItems: Array<T>) => void;
+  /** Filtering rule to be used */
+  filterRule: (item: T, query: string) => boolean;
+}
+
+class BaseFilterInput<T> extends Component<FilterInputProps<T>> {
+  render() {
+    const {
+      className,
+      inputContainerProps,
+      visualPlaceholder,
+      labelText,
+      labelMode,
+      optionalText,
+      status,
+      statusText,
+      id,
+      labelAlign,
+      'aria-describedby': ariaDescribedBy,
+      items: propItems,
+      onFiltering: propOnFiltering,
+      filterRule: propFilterRule,
+      ...passProps
+    } = this.props;
+
+    const onChangeHandler = (event: ChangeEvent<HTMLInputElement>) => {
+      const { items, onFiltering, filterRule } = this.props;
+      const { value } = event.target;
+
+      const filteredItems: T[] = items.reduce((filtered: T[], item: T) => {
+        if (filterRule(item, value)) {
+          filtered.push(item);
+        }
+        return filtered;
+      }, []);
+
+      onFiltering(filteredItems);
+    };
+
+    const statusTextId = statusText ? `${id}-statusText` : undefined;
+
+    return (
+      <HtmlDiv
+        {...inputContainerProps}
+        className={classnames(baseClassName, className, {
+          [filterInputClassNames.disabled]: !!passProps.disabled,
+          [filterInputClassNames.error]: status === 'error',
+        })}
+      >
+        <HtmlDiv
+          className={classnames(filterInputClassNames.wrapper, {
+            [filterInputClassNames.labelAlignLeft]: labelAlign === 'left',
+          })}
+        >
+          <LabelText
+            htmlFor={id}
+            as="label"
+            labelMode={labelMode}
+            optionalText={optionalText}
+          >
+            {labelText}
+          </LabelText>
+          <HtmlDiv className={filterInputClassNames.functionalityContainer}>
+            <HtmlDiv className={filterInputClassNames.inputElementContainer}>
+              <HtmlInput
+                {...passProps}
+                id={id}
+                className={filterInputClassNames.inputElement}
+                type="text"
+                placeholder={visualPlaceholder}
+                {...getConditionalAriaProp('aria-describedby', [
+                  statusTextId,
+                  ariaDescribedBy,
+                ])}
+                autoComplete="off"
+                aria-autocomplete="list"
+                auto-capitalize="false"
+                spellCheck="false"
+                onChange={onChangeHandler}
+              />
```

Comment:
This might benefit from aria role searchbox or listbox etc. Might require testing with screenreaders to determine which works the best. Had to add one for searchinput to make it work correctly at least.

Reply:
Probably the `role="listbox"` would be the closest one. But currently it causes confusion as screenreader then tells that user could start interacting with the items.

I suggest that we add this or some other role to **FilterInput** when the **Combobox** is builded to a state where it uses FilterInput component and is actually able to show the items in popover.

FilterInput will highly likely to see more changes also when implementing Combobox.